crates/store/re_types/definitions/rerun/blueprint/archetypes/dataframe_query.fbs

```diff
@@ -49,3 +50,38 @@ table DataframeQuery (
     time_range_queries: rerun.blueprint.components.TimeRangeQueries ("attr.rerun.component_optional", nullable,order: 500);
 }
 
+
+
+/// The query for the dataframe view.
+table DataframeQueryV2 (
+    "attr.rerun.scope": "blueprint"
+) {
+    // --- Optional ---
+
+    /// The timeline for this query.
+    ///
+    /// If unset, the time panel's timeline is used and stored.
+    timeline: rerun.blueprint.components.TimelineName ("attr.rerun.component_optional", nullable, order: 100);
+
+    /// If set, a range filter is applied.
+    ///
+    /// Note: will be unset as soon as `timeline` is changed.
+    //TODO(ab): should this use the list-with-timelines trick?
+    range_filter: rerun.blueprint.components.RangeFilter ("attr.rerun.component_optional", nullable, order: 200);
+
+    /// Whether the filter by event feature is active.
+    // Note: we use an explicit field here such that the filter column is not lost when the filter is deactivated.
+    filter_by_event_active: rerun.blueprint.components.FilterByEventActive ("attr.rerun.component_optional", nullable, order: 300);
+
+    /// The column used when the filter by event feature is used.
+    ///
+    /// Note: only valid if the entity/component exists on `timeline`.
+    filter_by_event_column: rerun.blueprint.components.ComponentColumnSelector ("attr.rerun.component_optional", nullable, order: 400);
```

Comment:
Note: in the blueprint API, the custom `__init__` will have a single `filter_by_event="/entity/path:ComponentName"` which will set both of these fields.